compiler/compile/OMRCompilation.cpp

```diff
@@ -698,16 +698,39 @@ bool OMR::Compilation::isPotentialOSRPointWithSupport(TR::TreeTop *tt)
 
    if (potentialOSRPoint && !self()->getOption(TR_FullSpeedDebug))
       {
-      // When in OSR HCR mode we need to make sure we check the BCI of the original
-      // call node to ensure we see the correct state of the doNotProfile flag
+
       if (self()->getHCRMode() == TR::osr &&
           (node->getOpCode().isCheck() || node->getOpCodeValue() == TR::treetop))
+         {
+         // When in OSR HCR mode we need to make sure we check the BCI of the original
+         // call node to ensure we see the correct state of the doNotProfile flag
          node = node->getFirstChild();
-      
-      TR_ByteCodeInfo &bci = node->getByteCodeInfo();
-      TR::ResolvedMethodSymbol *method = bci.getCallerIndex() == -1 ?
-         self()->getMethodSymbol() : self()->getInlinedResolvedMethodSymbol(bci.getCallerIndex());
-      potentialOSRPoint = method->supportsInduceOSR(bci, tt->getEnclosingBlock(), NULL, self(), false);
+
+         // In OSR HCR mode, the OSR point for a call occurs on its first evaluation.
+         // Therefore, it is necessary to check if the call is anchored under a prior treetop.
+         if (node->getReferenceCount() > 1)
+            {
+            TR::TreeTop *cursor = tt->getPrevTreeTop();
+            while (cursor && cursor->getNode()->getOpCodeValue() != TR::BBStart)
```

Comment:
Nodes can be commoned across an extended basic block, rather than just a single basic block. While this pass may not be running after block extension today it's probably better to end the search at the extended block start treetop (which other local optimizations do use via an existing API).